### PR TITLE
vcf-report: use smart_open crate to enable remote --custom-js-files

### PR DESCRIPTION
**Pull request by dlaehnemann**

### CODE REVIEW

**Review by fxwiegand — CHANGES_REQUESTED** [review]
Great idea! Looks good to me. Only one suggestion regarding the newly added test above.

**Review by fxwiegand — APPROVED** [review]
Looks good!

File: tests/lib.rs [hunk 1]
```diff
@@ -219,6 +219,72 @@ fn test_vcf_report() {
     fs::remove_dir_all("tests/test-vcf-report").unwrap();
 }
 
+#[test]
```
[the remark this answers is not in the document]
Comment (dlaehnemann) [reply tests/lib.rs:222]: For the test, I have now switched to downloading an external `js`-file that is not included among the standard files and check that its content is correct with a local copy in`tests/report/`. That should do the trick, right? Or do you have further suggestions of what to test (inclusion somewhere else?).

[the remark this answers is not in the document]
Comment (fxwiegand) [reply tests/lib.rs:222]: Thats exactly what I had in mind yep! Thanks!

**Pull request closed without merging**